storage/merkle/node_test.go

```diff
@@ -69,19 +84,20 @@ func TestShortHash(t *testing.T) {
 		require.Equal(t, ref, hex.EncodeToString(s.Hash()))
 	})
 
-	t.Run("65536-bit path", func(t *testing.T) {
+	t.Run("maxKeyLenBits-bit path", func(t *testing.T) {
 		// as path, we just repeat the following 32 bytes 256 times
 		k, _ := hex.DecodeString("1b30482d4dc8c1a8d846d05765c03a33f0267b56b9a7be8defe38958f89c95fc")
-		path := make([]byte, 0, 8192)
-		for i := 1; i <= 256; i++ {
+		path := make([]byte, 0, maxKeyLength)
+		for i := 1; i <= maxKeyLength/len(k); i++ {
 			path = append(path, k...)
 		}
+		path = append(path, k[:maxKeyLength%len(k)]...)
 		// expected value from python reference implementation
-		ref := "b49541ce56e5cd207cac3a08821ddfc61fc5aaf8886907d226581a7c53a48827"
+		ref := "22eb614e20aae939ff38a0dced42fdebb95199e725f1c898e2c5141e7c9d6848"
 
 		s := short{
 			path:  path,
-			count: 65536,
+			count: maxKeyLenBits,
```

Comment:
⚠️ 
To me, it looks like we are testing a case here that is illegal in our storage model:
* the longest possible key in our storage model is 65528 bits or 8191 bytes. 
* While you construct a key with 655**28** bytes: https://github.com/onflow/flow-go/blob/8f41c5c459f4be76e3afb64f4b11bbfd5b2d905d/storage/merkle/node_test.go#L90-L94 you set the bit length of the key to 655**35**: https://github.com/onflow/flow-go/blob/8f41c5c459f4be76e3afb64f4b11bbfd5b2d905d/storage/merkle/node_test.go#L100

```suggestion
			count: len(path),
```

Reply:
Good catch, I think this is solved now by fixing `maxKeyLenBits` to 65528 in [a21444e](https://github.com/onflow/flow-go/pull/1909/commits/a21444e0faada936ce5994b66642e5fa7c8ad414)